processor/processor.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"context"
+
+	"github.com/census-instrumentation/opencensus-service/data"
+)
+
+// MetricsDataProcessor is an interface that receives data.MetricsData, process it as needed, and
+// sends it to the next processing node if any or to the destination.
+//
+// ProcessMetricsData receives data.MetricsData for processing by the MetricsDataProcessor.
+type MetricsDataProcessor interface {
+	ProcessMetricsData(ctx context.Context, md data.MetricsData) error
+}
+
+// TraceDataProcessor is an interface that receives data.TraceData, process it as needed, and
+// sends it to the next processing node if any or to the destination.
+//
+// ProcessTraceData receives data.TraceData for processing by the TraceDataProcessor.
+type TraceDataProcessor interface {
+	ProcessTraceData(ctx context.Context, td data.TraceData) error
```

Comment:
How do you feel about making the proto model the "core" data model for the unified interface? Currently TraceData is an identical structure, and forces us to do an additional mapping.

Reply:
I think it is better to use [Trace|Metrics]Data because in the future we can add extra informations to these like "source" or "num_filtered".